src/contrib/mlir/core/pass/ng_dialect_fused_ops.cpp

```diff
@@ -0,0 +1,144 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+// NOTE: This file follows nGraph format style.
+// Follows nGraph naming convention for public APIs only, else MLIR naming convention.
+
+#include "ng_dialect_fused_ops.hpp"
+#include "contrib/mlir/core/ngraph_dialect/dialect.hpp"
+#include "contrib/mlir/core/ngraph_dialect/ops.hpp"
+#include "contrib/mlir/core/ngraph_dialect/type.hpp"
+
+#include <llvm/ADT/DenseSet.h>
+#include <llvm/ADT/STLExtras.h>
+#include <llvm/Analysis/TargetTransformInfo.h>
+#include <llvm/ExecutionEngine/Orc/JITTargetMachineBuilder.h>
+#include <llvm/IR/Module.h>
+#include <llvm/Support/ErrorOr.h>
+#include <llvm/Support/MemoryBuffer.h>
+#include <llvm/Support/SourceMgr.h>
+#include <llvm/Support/TargetSelect.h>
+#include <llvm/Target/TargetMachine.h>
+#include <mlir/Conversion/LoopToStandard/ConvertLoopToStandard.h>
+#include <mlir/Conversion/StandardToLLVM/ConvertStandardToLLVM.h>
+#include <mlir/Conversion/StandardToLLVM/ConvertStandardToLLVMPass.h>
+#include <mlir/Dialect/LLVMIR/LLVMDialect.h>
+#include <mlir/EDSC/Builders.h>
+#include <mlir/EDSC/Helpers.h>
+#include <mlir/EDSC/Intrinsics.h>
+#include <mlir/ExecutionEngine/ExecutionEngine.h>
+#include <mlir/ExecutionEngine/OptUtils.h>
+#include <mlir/IR/AffineExpr.h>
+#include <mlir/IR/IntegerSet.h>
+#include <mlir/IR/MLIRContext.h>
+#include <mlir/IR/StandardTypes.h>
+#include <mlir/Pass/PassManager.h>
+#include <mlir/Target/LLVMIR.h>
+#include <mlir/Transforms/DialectConversion.h>
+#include <mlir/Transforms/Passes.h>
+
+#include <iostream>
+#define DEBUG_TYPE "mlir-compiler"
+#define PASS_NAME "ngraph_dialect_fusion"
+
+using llvm::SmallVector;
+using llvm::StringRef;
+using llvm::ArrayRef;
+
+using namespace ngraph;
+using namespace mlir;
+using namespace mlir::edsc;
+using namespace mlir::edsc::op;
+
+namespace mlir
+{
+    static Value createSgemmOp(
+        PatternRewriter& rewriter, Operation* old_op, Value input1, Value input2, Value input3)
+    {
+        auto castedOp0 = dyn_cast_or_null<NGAddOp>(old_op);
+        (void)castedOp0;
+        SmallVector<Value, 4> values;
+        (void)values;
+        SmallVector<NamedAttribute, 4> attrs;
+        (void)attrs;
```

Comment:
are void castings on castedOp0, values, attrs,  types, etc. necessary?

Reply:
will remove, looks like it was added in the generated code to suppress compiler warnings ..i simply ported those.